src/main/scala/mesosphere/marathon/storage/migration/Migration.scala

```diff
@@ -100,15 +100,22 @@ class Migration(
 
   @SuppressWarnings(Array("all")) // async/await
   def migrateAsync(): Future[Seq[StorageVersion]] = async {
+
     val config = await(runtimeConfigurationRepository.get()).getOrElse(RuntimeConfiguration())
     // before backup/restore called, reset the runtime configuration
     await(runtimeConfigurationRepository.store(RuntimeConfiguration(None, None)))
     // step 1: backup current zk state
     await(config.backup.map(uri => backup.backup(new URI(uri))).getOrElse(Future.successful(Done)))
     // step 2: restore state from given backup
     await(config.restore.map(uri => backup.restore(new URI(uri))).getOrElse(Future.successful(Done)))
+
+    // mark migration as started
+    await(persistenceStore.startMigration())
     // last step run the migration, to ensure we can operate on the zk state
     val result = await(migrateStorage(backupCreated = config.backup.isDefined || config.restore.isDefined))
+    // mark migration as completed
+    await(persistenceStore.endMigration())
```

Comment:
There are some migrations that legitimately abort due to a configuration issue. IE https://github.com/mesosphere/marathon/blob/master/src/main/scala/mesosphere/marathon/storage/migration/MigrationTo15.scala#L64

Perhaps we should introduce a MigrationCancelled exception or something? And catch and endMigration ?

Reply:
A very valid point. Thanks, Tim. I will do it.